### PR TITLE
Add (very) basic test for -gsplit-dwarf

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

File: tests/test_other.py [hunk 1]
```diff
@@ -9793,3 +9793,10 @@ def test_split_module(self):
     result = self.run_js('test_split_module.js')
     self.assertNotIn('writing profile', result)
     self.assertIn('Hello! answer: 42', result)
+
+  def test_split_dwarf(self):
+    self.emcc_args += ['-gsplit-dwarf']
+    self.run_process([EMCC, '-c', path_from_root('tests', 'core', 'test_hello_world.c'), '-g', '-gsplit-dwarf'])
+    self.assertExists('test_hello_world.o')
+    # TODO(sbc): test the dwo is correct/usable
```
Comment (dschuff) [inline tests/test_other.py:9801]: could we dump it with `llvm-objdump` and assert something about the output here?

Comment (sbc100) [reply tests/test_other.py:9801]: Done

File: tests/test_other.py [hunk 2]
```diff
@@ -9872,3 +9884,19 @@ def test_split_main_module(self):
...
+    self.assertExists('test_hello_world.o')
+    self.assertExists('test_hello_world.dwo')
+    obj_out = read_sections_using_objdump('test_hello_world.o')
+    dwo_out = read_sections_using_objdump('test_hello_world.dwo')
+    obj_info = [s for s in obj_out if s[1] == '.debug_info'][0]
+    dwo_info = [s for s in dwo_out if s[1] == '.debug_info.dwo'][0]
+    obj_size = int(obj_info[2], 16)
+    dwo_size = int(dwo_info[2], 16)
+    print('.debug_info: %d' % obj_size)
+    print('.debug_info.dwo: %d' % dwo_size)
```
Comment (dschuff) [inline tests/test_other.py:9901]: Do we want to keep these print statements?

File: tests/test_other.py [hunk 3]
```diff
@@ -9872,3 +9884,19 @@ def test_split_main_module(self):
...
+    self.assertGreater(dwo_size, obj_size)
```
Comment (dschuff) [inline tests/test_other.py:9902]: Actually I guess if we're going so far as to look at the sections we could run `llvm-dwarfdump'. But this seems ok for now, better than nothing.

**Review by dschuff — APPROVED** [review]

**Comment by dschuff** [comment]
Let's not close this yet; we should probably have some kind of test for gsplit-dwarf.

**Comment by sbc100** [comment]
Has other tested landed in the last few years?  Can we close this one now?

**Comment by dschuff** [comment]
Yeah, I think other.test_split_dwarf_dwp covers what this test does.

**Pull request closed without merging**